src/main/java/com/mathworks/ci/Utilities.java

```diff
@@ -23,4 +33,47 @@ public static String getCellArrayFrmList(List<String> listOfStr){
         filteredListOfStr.replaceAll(val -> "'" + val.replaceAll("'", "''") + "'");
         return "{" + String.join(",", filteredListOfStr) + "}";
     }
+
+    public static void addMatlabToEnvPathFrmAxis(Computer cmp, TaskListener listener, EnvVars env)
+            throws IOException, InterruptedException {
+        String name = env.get(Message.getValue("Axis.matlab.key"));
+
+        // If no MATLAB axis is set or if 'Use MATLAB version' is selected, return
+        if (name == null || name.isEmpty() || env.get("matlabroot") != null){
+            return;
+        }
+
+        String matlabExecutablePath = getNodeSpecificHome(name, cmp.getNode(), listener, env) + "/bin";
+        env.put("PATH+matlabroot", matlabExecutablePath);
+        // Specify which MATLAB was added to path.
+        listener.getLogger().println("\n" + String.format(Message.getValue("matlab.added.to.path.from"), matlabExecutablePath) + "\n");
+    }
+
+    public static String getNodeSpecificHome(String instName, Node node, TaskListener listener, EnvVars env)
+            throws IOException, InterruptedException {
+        MatlabInstallation inst = MatlabInstallation.getInstallation(instName);
+        if(inst != null) {
+            if (node != null) {
```

Comment:
If both node and env are not null, then value of inst computed in the first if, will be immediately overwritten by the next if. Consider refactoring.

Reply:
This is by design to get Installation for specific node and then specific environment. Updated the code to be more readable